pilot/lib/pilot/manager.rb

```diff
@@ -13,7 +13,11 @@ class Manager
     def start(options, should_login: true)
       return if @config # to not login multiple times
       @config = options
-      login if should_login
+
+      # we will always start with App Store Connect API login if required and possible
+      # else fallback to 'should_login' param for 'apple_id' login
+      should_login_for_appstore_connect_api = Spaceship::ConnectAPI.token.nil? && (options[:api_key_path] || options[:api_key])
+      login if should_login_for_appstore_connect_api || should_login
```

Comment:
All existing unit-tests are passing and I will be adding more unit-tests to cover this logic after Josh's thoughts on this change. 

Changing something in `pilot start` already making me a little nervous 😇